src/main/java/software/amazon/awssdk/crt/s3/S3TcpKeepAliveOptions.java

```diff
@@ -0,0 +1,40 @@
+package software.amazon.awssdk.crt.s3;
+
+/**
+ * This class provides access to setting Tcp Keep Alive Options.
+ * If interval or timeout are zero, then default values are used.
+ */
+public class S3TcpKeepAliveOptions {
```

Comment:
Does this really need to be a separate class?

Reply:
There is a `SocketOptions` that has the tcp keep alive settings. This `S3TcpKeepAliveOptions` seems like a unnecessary complexity to me, it's not doing anything special to S3. 

I don't believe we need this in C [here](https://github.com/awslabs/aws-c-s3/pull/204/files#diff-e45799b2b3d8b2c5e2bba0698cb763ed106c53e07f3c5c049f4b2eee58c26382R155) as well. I'd use socket options instead.

Reply:
SocketOptions has additional properties that we don't want to allow override.